Testscripts/Windows/STRESSTEST-VERIFY-RESTART.ps1

```diff
@@ -17,54 +17,107 @@ function Main {
-            $isRestarted = $TestProvider.RestartAllDeployments($allVMData)
-            if ($isRestarted) {
-                Write-LogInfo "Virtual machine restart successful."
-                $testResult = "PASS"
-            } else {
-                Write-LogErr "Virtual machine restart failed."
-                $testResult = "FAIL"
-                break
-            }
-        } catch {
-            $ErrorMessage =  $_.Exception.Message
-            $ErrorLine = $_.InvocationInfo.ScriptLineNumber
-            Write-LogInfo "EXCEPTION : $ErrorMessage at line: $ErrorLine"
+    $expectedcount = 0
+    switch ($testType) {
+        {($_ -imatch "NVME")} {
+            $keyWord = "Non-Volatile memory controller";
+            $expectedcount = Get-ExpectedCount -ip $AllVmData.PublicIP -port $AllVmData.SSHPort -username $user -password $password -size $AllVmData.InstanceSize -type "NVME"
             break
-        } finally {
-            if (-not $testResult) {
-                $testResult = "Aborted"
+        }
+        {($_ -imatch "GPU")} {
+            $keyWord = "NVIDIA";
+            $expectedcount = Get-ExpectedCount -ip $AllVmData.PublicIP -port $AllVmData.SSHPort -username $user -password $password -size $AllVmData.InstanceSize -type "GPU"
+            break
+        }
+        {($_ -imatch "SRIOV")} {
+            $keyWord = "Mellanox"
+            foreach($nic in (Get-AzVM -Name $AllVmData.RoleName).NetworkProfile.NetworkInterfaces) {
+                if((Get-AzNetworkInterface -ResourceId $nic.Id).EnableAcceleratedNetworking) {
+                    $expectedcount++
+                }
             }
-            $resultArr += $testResult
+            break
         }
+        default { }
     }
-    $rebootNr--
-    Write-LogInfo "Reboot Stress Test Result: $rebootNr/$rebootNumber"
-    if (($rebootNr - 1) -lt $rebootNumber) {
+    Run-LinuxCmd -ip $AllVmData.PublicIP -port $AllVmData.SSHPort -username $user -password $password -command "which lspci || (. ./utils.sh && install_package pciutils)" -runAsSudo | Out-Null
+    $noIssues = Check-KernelLogs -allVMData $AllVmData
+    if (!$noIssues) {
+        $CurrentTestResult.TestSummary += New-ResultSummary -testResult "FAIL" `
+            -metaData "FirstBoot : Call Trace Verification" -checkValues "PASS,FAIL,ABORTED" -testName $currentTestData.testName
         $testResult = "FAIL"
+    } else {
+        for ($rebootNr = 1; $rebootNr -le $rebootNumber; $rebootNr++) {
+            try {
+                Write-LogInfo ("Trying to restart {0}: {1} / {2} ..." `
+                    -f @($AllVMData.RoleName, $rebootNr, $rebootNumber))
+                $isRestarted = $TestProvider.RestartAllDeployments($AllVmData)
+                if ($isRestarted) {
+                    Write-LogInfo "Virtual machine restart successful."
+                    $noIssues = Check-KernelLogs -allVMData $AllVmData
+                    if (!$noIssues) {
```

Comment:
rename the variable, as it's not intuitive and  reader friendly, especially using a condition to do logic if ... else...,